qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +98,327 @@ class Command(BaseModel):
+                 pulse_library: List[PulseLibraryItem],
+                 cmd_def: List[Command],
+                 **kwargs: Dict[str, Any]):
+        """
+        Validate and reformat transport layer inputs to initialize.
+
+        Args:
+            qubit_freq_est: Estimated qubit frequencies in GHz.
+            meas_freq_est: Estimated measurement cavity frequencies in GHz.
+            buffer: Default buffer time (in units of dt) between pulses.
+            pulse_library: Pulse name and sample definitions.
+            cmd_def: Operation name and definition in terms of Commands.
+            **kwargs: Other attributes for the super class.
+        """
+        super().__init__(**kwargs)
+
         self.buffer = buffer
+        self.qubit_freq_est = [freq * 1e9 for freq in qubit_freq_est]
+        self.meas_freq_est = [freq * 1e9 for freq in meas_freq_est]
         self.pulse_library = pulse_library
         self.cmd_def = cmd_def
+        self.ops_def = InstructionScheduleMap()
+
+        self.converter = QobjToInstructionConverter(pulse_library)
+        for op in cmd_def:
+            pulse_insts = [self.converter(inst) for inst in op.sequence]
+            schedule = ParameterizedSchedule(*pulse_insts, name=op.name)
+            self.ops_def.add(op.name, op.qubits, schedule)
+
+    def __str__(self):
+        qubit_freqs = [freq / 1e9 for freq in self.qubit_freq_est]
+        meas_freqs = [freq / 1e9 for freq in self.meas_freq_est]
+        qfreq = "Qubit Frequencies [GHz]\n{freqs}".format(freqs=qubit_freqs)
+        mfreq = "Measurement Frequencies [GHz]\n{freqs} ".format(freqs=meas_freqs)
+        return ("<{name}({ops}{qfreq}\n{mfreq})>"
+                "".format(name=self.__class__.__name__, ops=str(self.ops_def),
+                          qfreq=qfreq, mfreq=mfreq))
+
+    def build_cmd_def(self) -> 'InstructionScheduleMap':
+        """
+        Return the InstructionScheduleMap built for this PulseDefaults instance.
 
-        super().__init__(**kwargs)
+        Returns:
+            InstructionScheduleMap: `InstructionScheduleMap` instance generated from defaults
+        """
+        warnings.warn("This method is deprecated. Returning a InstructionScheduleMap instead. "
+                      "This can be accessed simply through the `ops_def` attribute of this "
+                      "PulseDefaults instance.", DeprecationWarning)
+        return self.ops_def
+
+
+class InstructionScheduleMap():
+    """Mapping from QuantumCircuit Instructions to Schedules."""
+
+    def __init__(self):
+        """
+        Initialize an instruction -> schedule mapper. May take transport layer `backend.defaults()`
+        args to initialize with backend data.
+        """
+        # The processed and reformatted circuit operation definitions
+        self._ops_def = defaultdict(dict)
+        # A backwards mapping from qubit to supported operation
+        self._qubit_ops = defaultdict(set)
+
+    def ops(self) -> List[str]:
```

Comment:
Could this be a property?

Reply:
sure

Reply:
oh, I made it not a property to be consistent with `cmds()` not being a property. It has to migrate to `instructions` anyway, so I'm ok with it if you don't think it matters

Reply:
I don't think it matters, might be nicer.